sources/reference/usage/time-constraints.md

```diff
@@ -0,0 +1,12 @@
+# Adding Time Constraints to an Experiment
+
+It is a common requirement to execute a chaos experiment for a certain period of
+ time, ending the experiment if it goes on indefinitely.
+
+We've been very careful to rely on other tools for these types of concerns, and 
+so timing constraints are not a built-in feature of the Chaos Toolkit's 
+experiments. 
+
+Typically we have used a  [Kubernetes job](https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/) 
```

Comment:
I think this goes with Scheduling actually as the Kube job is meant a distributed cron so to speak :p

sorry :/

Reply:
No worries at all. But does that mean we don't have an answer (yet) for time constraints?

Reply:
Ah sorry, I didn't realise that.

The current support we have is only a timeout on a per activity basis (only for process and HTTP providers though, meaning a rogue Python extension would have to be killed by the operator).

This could require an issue for a proper discussion/solution, wdyt?

Reply:
I think this is a new feature now really, so an issue for now I think.